llvm/lib/ObjCopy/ELF/ELFObject.cpp

```diff
@@ -2847,24 +2847,24 @@ void SRECSectionWriter::writeRecord(SRecord &Record, uint64_t Off) {
   memcpy(Out.getBufferStart() + Off, Data.data(), Data.size());
 }
 
-void SRECSectionWriter::writeRecords(uint32_t Entry) {
+void SRECSizeCalculator::writeRecords(uint32_t Entry) {
+  // The ELF header could contain an entry point outside of the sections we have
+  // seen that does not fit the current record Type.
   Type = std::max(Type, SRecord::getType(Entry));
   uint64_t Off = HeaderSize;
   for (SRecord &Record : Records) {
     Record.Type = Type;
-    writeRecord(Record, Off);
     Off += Record.getSize();
   }
   Offset = Off;
 }
 
-void SRECSectionWriterBase::writeRecords(uint32_t Entry) {
-  // The ELF header could contain an entry point outside of the sections we have
-  // seen that does not fit the current record Type.
+void SRECSectionWriter::writeRecords(uint32_t Entry) {
   Type = std::max(Type, SRecord::getType(Entry));
```

Comment:
Could we move this function into the base class, and just make `writeRecord` the sub-class specific method (in the size calculation case, it does nothing)?

Reply:
Done